Compilation warnings raised on Ubuntu 18.04

<!-- Please search existing issues for potential duplicates before filing yours:
https://github.com/godotengine/godot/issues?q=is%3Aissue
-->

**Godot version:**
<!-- Specify commit hash if non-official. -->
commit 46bff1ab54462c34c808528b7fddc24359d81ad3 (HEAD -> master, origin/master, origin/HEAD)
Date:   Mon Jun 17 15:07:27 2019 +0200

**OS/device including version:**
<!-- Specify GPU model and drivers if graphics-related. -->
$ cat /etc/lsb-release
```
DISTRIB_ID=Ubuntu
DISTRIB_RELEASE=18.04
DISTRIB_CODENAME=bionic
DISTRIB_DESCRIPTION="Ubuntu 18.04.2 LTS"
```

**Issue description:**
<!-- What happened, and what was expected. -->
Our builder now stops for the tools **relaes_debug** target (the debug target still works):
`scons platform=x11 bits=64 dev=yes -j6 target=release_debug`
or
`scons CC="gcc-8" CXX="g++-8" platform=x11 bits=64 dev=yes -j6 target=release_debug`

```
scons: Reading SConscript files ...
Enabling ALSA
Enabling PulseAudio
Scons cache enabled... (path: '/df/scons-cache')
Checking for C header file mntent.h... yes
Current cache limit is 11.72 GB (used: 1.05 GB)
scons: done reading SConscript files.
scons: Building targets ...
[  1%] g++-8 -o platform/x11/godot_x11.x11.opt.tools.64.o -c -Wctor-dtor-privacy -Wnon-virtual-dtor -Wnoexcept -Wplacement-new=1 -g1 -O2 -pipe -fpie -Wall -Wextra -Wno-unused-parameter -Wwrite-strings -Wshadow-local -Wno-clobbered -Walloc-zero -Wduplicated-branches -Wduplicated-cond -Wstringop-overflow=4 -Wlogical-op -Werror -DDEBUG_ENABLED -DTOUCH_ENABLED -DALSA_ENABLED -DALSAMIDI_ENABLED -DPULSEAUDIO_ENABLED -DJOYDEV_ENABLED -DX11_ENABLED -DUNIX_ENABLED -DOPENGL_ENABLED -DGLES_ENABLED -DZSTD_STATIC_LINKING_ONLY -DGLAD_ENABLED -DGLES_OVER_GL -DFREETYPE_ENABLED -DSVG_ENABLED -DNO_EDITOR_SPLASH -D_REENTRANT -DPTRCALL_ENABLED -DTOOLS_ENABLED -DGDSCRIPT_ENABLED -DMINIZIP_ENABLED -DHAVE_MNTENT -Ithirdparty/nanosvg -Ithirdparty/glad -Ithirdparty/zstd -Iplatform/x11 -I. -Ieditor -I/usr/include/freetype2 -I/usr/include/libpng16 platform/x11/godot_x11.cpp
[  1%] g++-8 -o platform/x11/context_gl_x11.x11.opt.tools.64.o -c -Wctor-dtor-privacy -Wnon-virtual-dtor -Wnoexcept -Wplacement-new=1 -g1 -O2 -pipe -fpie -Wall -Wextra -Wno-unused-parameter -Wwrite-strings -Wshadow-local -Wno-clobbered -Walloc-zero -Wduplicated-branches -Wduplicated-cond -Wstringop-overflow=4 -Wlogical-op -Werror -DDEBUG_ENABLED -DTOUCH_ENABLED -DALSA_ENABLED -DALSAMIDI_ENABLED -DPULSEAUDIO_ENABLED -DJOYDEV_ENABLED -DX11_ENABLED -DUNIX_ENABLED -DOPENGL_ENABLED -DGLES_ENABLED -DZSTD_STATIC_LINKING_ONLY -DGLAD_ENABLED -DGLES_OVER_GL -DFREETYPE_ENABLED -DSVG_ENABLED -DNO_EDITOR_SPLASH -D_REENTRANT -DPTRCALL_ENABLED -DTOOLS_ENABLED -DGDSCRIPT_ENABLED -DMINIZIP_ENABLED -DHAVE_MNTENT -Ithirdparty/nanosvg -Ithirdparty/glad -Ithirdparty/zstd -Iplatform/x11 -I. -Ieditor -I/usr/include/freetype2 -I/usr/include/libpng16 platform/x11/context_gl_x11.cpp
[  1%] g++-8 -o platform/x11/crash_handler_x11.x11.opt.tools.64.o -c -Wctor-dtor-privacy -Wnon-virtual-dtor -Wnoexcept -Wplacement-new=1 -g1 -O2 -pipe -fpie -Wall -Wextra -Wno-unused-parameter -Wwrite-strings -Wshadow-local -Wno-clobbered -Walloc-zero -Wduplicated-branches -Wduplicated-cond -Wstringop-overflow=4 -Wlogical-op -Werror -DDEBUG_ENABLED -DTOUCH_ENABLED -DALSA_ENABLED -DALSAMIDI_ENABLED -DPULSEAUDIO_ENABLED -DJOYDEV_ENABLED -DX11_ENABLED -DUNIX_ENABLED -DOPENGL_ENABLED -DGLES_ENABLED -DZSTD_STATIC_LINKING_ONLY -DGLAD_ENABLED -DGLES_OVER_GL -DFREETYPE_ENABLED -DSVG_ENABLED -DNO_EDITOR_SPLASH -D_REENTRANT -DPTRCALL_ENABLED -DTOOLS_ENABLED -DGDSCRIPT_ENABLED -DMINIZIP_ENABLED -DHAVE_MNTENT -Ithirdparty/nanosvg -Ithirdparty/glad -Ithirdparty/zstd -Iplatform/x11 -I. -Ieditor -I/usr/include/freetype2 -I/usr/include/libpng16 platform/x11/crash_handler_x11.cpp
platform/x11/godot_x11.cpp: In function 'int main(int, char**)':
platform/x11/godot_x11.cpp:59:8: error: ignoring return value of 'int chdir(const char*)', declared with attribute warn_unused_result [-Werror=unused-result]
   chdir(cwd);
   ~~~~~^~~~~
cc1plus: all warnings being treated as errors
scons: *** [platform/x11/godot_x11.x11.opt.tools.64.o] Error 1
scons: building terminated because of errors.
```

$ gcc -v

```
Using built-in specs.
COLLECT_GCC=gcc
COLLECT_LTO_WRAPPER=/usr/lib/gcc/x86_64-linux-gnu/7/lto-wrapper
OFFLOAD_TARGET_NAMES=nvptx-none
OFFLOAD_TARGET_DEFAULT=1
Target: x86_64-linux-gnu
Configured with: ../src/configure -v --with-pkgversion='Ubuntu 7.4.0-1ubuntu1~18.04.1' --with-bugurl=file:///usr/share/doc/gcc-7/README.Bugs --enable-languages=c,ada,c++,go,brig,d,fortran,objc,obj-c++ --prefix=/usr --with-gcc-major-version-only --program-suffix=-7 --program-prefix=x86_64-linux-gnu- --enable-shared --enable-linker-build-id --libexecdir=/usr/lib --without-included-gettext --enable-threads=posix --libdir=/usr/lib --enable-nls --with-sysroot=/ --enable-clocale=gnu --enable-libstdcxx-debug --enable-libstdcxx-time=yes --with-default-libstdcxx-abi=new --enable-gnu-unique-object --disable-vtable-verify --enable-libmpx --enable-plugin --enable-default-pie --with-system-zlib --with-target-system-zlib --enable-objc-gc=auto --enable-multiarch --disable-werror --with-arch-32=i686 --with-abi=m64 --with-multilib-list=m32,m64,mx32 --enable-multilib --with-tune=generic --enable-offload-targets=nvptx-none --without-cuda-driver --enable-checking=release --build=x86_64-linux-gnu --host=x86_64-linux-gnu --target=x86_64-linux-gnu
Thread model: posix
gcc version 7.4.0 (Ubuntu 7.4.0-1ubuntu1~18.04.1)
```

$ gcc-8 -v
```
Using built-in specs.
COLLECT_GCC=gcc-8
COLLECT_LTO_WRAPPER=/usr/lib/gcc/x86_64-linux-gnu/8/lto-wrapper
OFFLOAD_TARGET_NAMES=nvptx-none
OFFLOAD_TARGET_DEFAULT=1
Target: x86_64-linux-gnu
Configured with: ../src/configure -v --with-pkgversion='Ubuntu 8.3.0-6ubuntu1~18.04.1' --with-bugurl=file:///usr/share/doc/gcc-8/README.Bugs --enable-languages=c,ada,c++,go,brig,d,fortran,objc,obj-c++ --prefix=/usr --with-gcc-major-version-only --program-suffix=-8 --program-prefix=x86_64-linux-gnu- --enable-shared --enable-linker-build-id --libexecdir=/usr/lib --without-included-gettext --enable-threads=posix --libdir=/usr/lib --enable-nls --enable-bootstrap --enable-clocale=gnu --enable-libstdcxx-debug --enable-libstdcxx-time=yes --with-default-libstdcxx-abi=new --enable-gnu-unique-object --disable-vtable-verify --enable-libmpx --enable-plugin --enable-default-pie --with-system-zlib --with-target-system-zlib --enable-objc-gc=auto --enable-multiarch --disable-werror --with-arch-32=i686 --with-abi=m64 --with-multilib-list=m32,m64,mx32 --enable-multilib --with-tune=generic --enable-offload-targets=nvptx-none --without-cuda-driver --enable-checking=release --build=x86_64-linux-gnu --host=x86_64-linux-gnu --target=x86_64-linux-gnu
Thread model: posix
gcc version 8.3.0 (Ubuntu 8.3.0-6ubuntu1~18.04.1) 
```

--- a/platform/x11/godot_x11.cpp
+++ b/platform/x11/godot_x11.cpp
@@ -55,8 +55,11 @@ int main(int argc, char *argv[]) {
 		os.run(); // it is actually the OS that decides how to run
 	Main::cleanup();
 
-	if (ret)
-		chdir(cwd);
+	if (ret) { // Previous getcwd was successful
+		if (chdir(cwd) != 0) {
+			ERR_PRINT("Couldn't return to previous working directory.");
+		}
+	}
 	free(cwd);
 
 	return os.get_exit_code();

